components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodStartupActivity.kt

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package io.gitpod.jetbrains.remote
+
+import com.intellij.diagnostic.VMOptions
+import com.intellij.ide.BrowserUtil
+import com.intellij.ide.actions.OpenFileAction
+import com.intellij.notification.NotificationAction
+import com.intellij.notification.NotificationGroupManager
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.startup.StartupActivity
+import com.intellij.openapi.util.BuildNumber
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiFileFactory
+import com.intellij.psi.PsiManager
+import com.intellij.util.application
+import io.gitpod.jetbrains.remote.utils.GitpodConfig
+import io.gitpod.jetbrains.remote.utils.GitpodConfig.YamlKey.jetbrains
+import io.gitpod.jetbrains.remote.utils.GitpodConfig.YamlKey.vmOptions
+import io.gitpod.jetbrains.remote.utils.GitpodConfig.defaultXmxOption
+import io.gitpod.jetbrains.remote.utils.GitpodConfig.gitpodYamlFile
+import io.gitpod.jetbrains.remote.utils.GitpodConfig.gitpodYamlReferenceLink
+import org.jetbrains.yaml.YAMLFileType
+import org.jetbrains.yaml.YAMLUtil
+import org.jetbrains.yaml.psi.YAMLFile
+import org.jetbrains.yaml.psi.YAMLKeyValue
+import java.nio.file.Paths
+
+class GitpodStartupActivity : StartupActivity.Background {
```

Comment:
We should perform validation whenever content of .gitpod.yml is changing even if dirty. Not only once. Should not it be rather local inspection?